clients/drcachesim/common/trace_entry.h

```diff
@@ -906,6 +909,19 @@ typedef enum {
      * The initial part can be used by a simulator for warmup.
      */
     OFFLINE_FILE_TYPE_BIMODAL_FILTERED_WARMUP = 0x2000,
+    /**
+     * Indicates an offline trace that contains trace templates for some system calls.
+     * The individual traces are separated by a #TRACE_MARKER_TYPE_SYSCALL marker which
+     * also specifies what system call the following trace belongs to. This file can be
+     * used with -syscall_template_file to raw2trace to create a
+     * #OFFLINE_FILE_TYPE_KERNEL_SYSCALLS trace. See the sample file written by the
+     * burst_syscall_inject.cpp test for more details on the expected format for the
+     * system call template file.
```

Comment:
Probably best to put a notice that the template file format may change (e.g., we may no longer support a single concatenated file and need to move to separate files if the approach of reading everything into memory proves untenable).

Reply:
Because zip support is assumed to be optional in drmemtrace, won't it be better if we continue supporting non-zipfile templates? If the template file becomes too large for raw2trace, it'd be up to the user to enable zip support during build and generate the zip file format template file.

Reply:
How would just switching to zip help with a too-large file?  The reading code here would have to be changed to not read into memory up front.  If we do that I think we should not support a concatenated file and either require zip for this feature or take in a list of files.

The other factor is that we want to check in to the repo some canonical traces for Linux syscalls.  I would think those should be in zip format or should be separate files (all the arguments about needing special tools to update when compressed).  So I don't think we can easily support a single file much longer than this initial PR -- I suggest we drop that support as soon as there's an alternative.

Reply:
>  If we do that I think we should not support a concatenated file and either require zip for this feature or take in a list of files.

I was thinking that we could continue supporting both. But maybe that's not worth it.